app/vmselect/netstorage/netstorage.go

```diff
@@ -3112,3 +3251,30 @@ func (pnc *perNodeCounter) GetTotal() uint64 {
 //
 // See https://github.com/golang/go/blob/704401ffa06c60e059c9e6e4048045b4ff42530a/src/runtime/malloc.go#L11
 const maxFastAllocBlockSize = 32 * 1024
+
+// execSearchQuery calls cb for with marshaled requestData for each tenant in sq.
+func execSearchQuery(qt *querytracer.Tracer, sq *storage.SearchQuery, cb func(qt *querytracer.Tracer, requestData []byte, t storage.TenantToken) any) []any {
+	if sq.IsMultiTenant && sq.TenantTokens == nil {
+		logger.Panicf("BUG: missing TenantTokens in multi-tenant search query")
```

Comment:
This assert looks a bit tricky to me.

 It's a common practice in go to use a nil slices.  And later after some refactoring TenantTokens could become nil slice.
 In this case vmselect may panic in runtime, if provided filters matches 0 tenants.

 I think, it's better to remove this assert.